zanata-common-api/src/main/java/org/zanata/common/DocumentType.java

```diff
@@ -41,37 +48,82 @@
+    PROPERTIES("properties"), PROPERTIES_UTF8("properties"),
+    XML("xml"), XLIFF("xml");
+
+    private static final Set<String> allSourceExtensions = buildExtensionsList(true);
 
-    private static final List<String> allExtensions = buildExtensionsList();
+    private static final Set<String> allTranslationExtensions = buildExtensionsList(false);
 
-    private static List<String> buildExtensionsList() {
-        List<String> allExtensions = new ArrayList<String>();
+    private static Map<String, String> buildPotMap() {
+        Map<String, String> potMap = new HashMap<>();
+        potMap.put("pot", "po");
+        return unmodifiableMap(potMap);
+    }
+
+    private static Set<String> buildExtensionsList(boolean source) {
+        Set<String> allExtensions = new HashSet<>();
         for (DocumentType type : DocumentType.values()) {
-            allExtensions.addAll(type.extensions);
+            if (source) {
+                allExtensions.addAll(type.getSourceExtensions());
+            } else {
+                allExtensions.addAll(type.getTranslationExtensions());
+            }
         }
-        return unmodifiableList(allExtensions);
+        return unmodifiableSet(allExtensions);
+    }
+
+    public static @Nullable DocumentType getByName(String name) {
+        try {
+            return DocumentType.valueOf(name.toUpperCase());
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    /**
+     * @return a read-only list of source file extensions for known file types
+     */
+    public static Set<String> getAllSourceExtensions() {
+        return allSourceExtensions;
     }
 
     /**
-     * @return a read-only list of file extensions for known file types
+     * @return a read-only list of translation file extensions for known file types
      */
-    public static List<String> getAllExtensions() {
-        return allExtensions;
+    public static Set<String> getAllTranslationExtensions() {
+        return allTranslationExtensions;
+    }
+
+    public static Set<DocumentType> fromSourceExtension(String sourceExtension) {
+        return fromExtension(true, sourceExtension);
     }
 
-    // FIXME damason: rename typeFor to fromString
-    public static DocumentType typeFor(String extension) {
+    public static Set<DocumentType> fromTranslationExtension(
+            String translationExtension) {
+        return fromExtension(false, translationExtension);
+    }
+
+    private static Set<DocumentType> fromExtension(boolean source,
+            String extension) {
+        Set<DocumentType> documentTypes = new HashSet<>();
         for (DocumentType type : DocumentType.values()) {
-            if (type.extensions.contains(extension)) {
-                return type;
+            Collection<String> extensions;
+            if (source) {
+                extensions = type.extensions.keySet();
+            } else {
+                extensions = type.extensions.values();
+            }
```

Comment:
I prefer a ternary expression for something like this because I find it easier to read, but that is just personal preference.

`Collection<String> extensions = source ? type.extensions.keySet() : type.extensions.values();`